source/Ship.cpp

```diff
@@ -4069,3 +4071,26 @@ void Ship::CreateSparks(vector<Visual> &visuals, const Effect *effect, double am
 			visuals.emplace_back(*effect, angle.Rotate(point) + position, velocity, angle);
 	}
 }
+
+
+
+void Ship::CalculateLandingSpeed()
+{
+	const float mass = Mass();
+	const float shipLandingSpeed = attributes.Get("landing speed");
+
+	const StellarObject *planet = landingPlanet->GetSystem()->FindStellar(landingPlanet);
+	const auto planetAttributes = landingPlanet->Attributes();
+	const float planetLandingSpeed = planetAttributes.count("very slow landing") ? .8f :
+		planetAttributes.count("slow landing") ? .9f :
+		(planet->IsMoon() || planetAttributes.count("fast landing")) ? 1.1f :
+		(planet->IsStation() || planetAttributes.count("very fast landing")) ? 1.2f :
+		1.f;
```

Comment:
Not totally a fan of reusing attributes like this. Having an double node on a `object` makes more sense I think? Thoughts?

Reply:
I'm not sure I follow what you mean by a `double node on an object`?
This is just a very small added feature and I tried to make it as simple as possible, it makes sense this would be linked to the attributes of the planet in my opinion.

Reply:
Why can't it be part of the planet, not as an attribute:

```
planet test
    "landing speed" "very slow"
```

Reply:
You're right, I'll change it to be that way